python/ml4ir/base/tests/test_auto_dag_network.py

```diff
@@ -123,4 +128,218 @@ def test_simple_topological(self):
+                            args:
+                              clip_value_min: 0.
+                              clip_value_max: 1000000.
+                    serving_info:
+                      name: pageViewsScore
+                    tfrecord_type: sequence
+                  - name: query_text
+                    node_name: query_text
+                    trainable: true
+                    dtype: string
+                    log_at_inference: true
+                    feature_layer_info:
+                      type: numeric
+                      shape: null
+                      fn: bytes_sequence_to_encoding_bilstm
+                      args:
+                        encoding_type: bilstm
+                        encoding_size: 128
+                        embedding_size: 128
+                        max_length: 20
+                    preprocessing_info:
+                      - fn: preprocess_text
+                        args:
+                          remove_punctuation: true
+                          to_lower: true
+                    serving_info:
+                      name: q
+                    tfrecord_type: context
+                  - name: Title
+                    node_name: Title
+                    trainable: true
+                    dtype: string
+                    log_at_inference: true
+                    feature_layer_info:
+                      type: numeric
+                      shape: null
+                      fn: bytes_sequence_to_encoding_bilstm
+                      args:
+                        encoding_type: bilstm
+                        encoding_size: 128
+                        embedding_size: 128
+                        max_length: 20
+                    preprocessing_info:
+                      - fn: preprocess_text
+                        args:
+                          remove_punctuation: true
+                          to_lower: true
+                    serving_info:
+                      name: Title
+                    tfrecord_type: context
+                """
+            ),
+            self.file_io.logger
+        )
+
+    def test_tie_weights(self):
+        model = AutoDagNetwork(model_config=self.model_config, feature_config=self.feat_config, file_io=self.file_io)
+        with self.subTest("Test layers are the same"):
+            self.assertEqual(model.model_graph.get_node("global1").layer, model.model_graph.get_node("global2").layer)
+        with self.subTest("Inputs should be different"):
+            self.assertNotEqual(model.model_graph.get_node("global1").inputs,
+                                model.model_graph.get_node("global2").inputs)
+
+    def test_without_tie_weights(self):
+        model_config = copy.deepcopy(self.model_config)
+        model_config.pop("tie_weights")
+        model = AutoDagNetwork(model_config=model_config, feature_config=self.feat_config, file_io=self.file_io)
+        self.assertIsNotNone(model.model_graph)
```

Comment:
@arvindsrikantan This feels like an insufficient test to me. Can we check if the actual connections match with the model config or something similar to tests for the graph?

Reply:
Thanks!